pkg/agent/flowexporter/flowrecords/flow_records.go

```diff
@@ -71,12 +71,23 @@ func (fr *FlowRecords) ValidateAndUpdateStats(connKey flowexporter.ConnectionKey
 }
 
 // ForAllFlowRecordsDo executes the callback for all records in the flow record map
-func (fr *FlowRecords) ForAllFlowRecordsDo(callback flowexporter.FlowRecordCallBack) error {
-	for k, v := range fr.recordsMap {
-		err := callback(k, v)
-		if err != nil {
-			klog.Errorf("Error when executing callback for flow record")
-			return err
+func (fr *FlowRecords) ForAllFlowRecordsDo(callback flowexporter.FlowRecordCallBack, v4 bool, v6 bool, v4ID uint16, v6ID uint16) error {
+	if v4 {
+		for k, v := range fr.recordsMap {
```

Comment:
could we add isv4 field to the `FlowRecord` struct? This can be populated through the call back function `addOrUpdateFlowRecord` when creating new record. This may save sending multiple args in flow_record.go functions and exporter.go functions including template IDs.

Reply:
Good idea. Updated. PTAL.